pkg/collectors/virtualservices.go

```diff
@@ -0,0 +1,110 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"github.com/douglas-reid/istio-state-metrics/pkg/apis/networking/v1alpha3"
+	"github.com/douglas-reid/istio-state-metrics/pkg/client/clientset/versioned"
+	informers "github.com/douglas-reid/istio-state-metrics/pkg/client/informers/externalversions/networking/v1alpha3"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descVirtualServiceHostsName          = "istio_pilot_virtual_service_hosts"
+	descVirtualServiceHostsHelp          = "Information about Hosts in Pilot VirtualServices"
+	descVirtualServiceHostsDefaultLabels = []string{"VirtualService", "hosts"}
```

Comment:
can we add `namespace` as a label as well?

Reply:
Yeah, I think it would be good. It would enable querying across namespaces.

So, the `virtual_service` label would be the short name and `namespace` would be the namespace. I believe in the code you are concat'ing the two. Instead of that, let's split them out into two labels.

and then let's use that split on *all* metrics in this PR.